runtime/gc_base/modronapi.cpp

```diff
@@ -586,7 +586,12 @@ j9gc_pool_maxmemory(J9JavaVM *javaVM, UDATA poolID)
 		maxsize = extensions->memoryMax;
 		break;
 	case J9_GC_MANAGEMENT_POOL_REGION_EDEN :
-		maxsize = extensions->tarokIdealEdenMaximumBytes;
+//		maxsize = extensions->tarokIdealEdenMaximumBytes;
+		if (((MM_GCExtensions *)extensions)->userSpecifiedParameters._Xmn._wasSpecified || ((MM_GCExtensions *)extensions)->userSpecifiedParameters._Xmnx._wasSpecified) {
```

Comment:
you don't need these casts to `(MM_GCExtensions *)` if you change cast at `extensions` definition line 543, function returns `MM_GCExtensions *` type any way